spec/rubocop/cop/rspec/named_subject_spec.rb

```diff
@@ -63,27 +63,47 @@ def foo
   context 'when IgnoreSharedExamples is false' do
     let(:cop_config) { { 'IgnoreSharedExamples' => false } }
 
-    it_behaves_like 'checking subject outside of shared examples'
+    it_behaves_like 'checking subject outside of shared examples', 'subject'
+    it_behaves_like 'checking subject outside of shared examples', 'subject!'
 
     it 'checks shared_examples for explicit subject usage' do
       expect_offense(<<-RUBY)
         RSpec.describe User do
           subject(:new_user) { described_class.new }
 
           shared_examples_for 'a valid new user' do
-            it "is a User" do
-              expect(subject).to be_a(User)
-                     ^^^^^^^ Name your test subject if you need to reference it explicitly.
+            context 'when subject' do
+              it "is a User" do
+                expect(subject).to be_a(User)
+                       ^^^^^^^ Name your test subject if you need to reference it explicitly.
+              end
+
+              it "is valid" do
+                expect(subject.valid?).to be(true)
+                       ^^^^^^^ Name your test subject if you need to reference it explicitly.
+              end
+
+              it "is new" do
+                expect(subject).to be_new_record
+                       ^^^^^^^ Name your test subject if you need to reference it explicitly.
+              end
             end
 
-            it "is valid" do
-              expect(subject.valid?).to be(true)
-                     ^^^^^^^ Name your test subject if you need to reference it explicitly.
-            end
-
-            it "is new" do
-              expect(subject).to be_new_record
-                     ^^^^^^^ Name your test subject if you need to reference it explicitly.
+            context 'when subject!' do
+              it "is a User" do
+                expect(subject!).to be_a(User)
```

Comment:
well, even if you define your subject with `subject!`, you'd still reference it as `subject` in tests.

Reply:
I may have misunderstood something, I separated `it`.

Reply:
If I have misunderstood, please point it out. I separated the `it` subject and made the `subject` and `subject!` explicit.

Reply:
I mean that you can't really use `subject!` in an expectation:

> Failure/Error: expect(subject!).to eq('foo')
       `subject!` is not available from within an example (e.g. an `it` block) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc). It is only available on an example group (e.g. a `describe` or `context` block).

You can still declare it as `subject!` or `subject!(:foo)`, but then in the actual test you'll be using `subject` or `foo`. So we don't need to change how we search for subject usage, but the declarations of a subject should use the language module, unless there is specific reason they want to ignore `subject!`

Reply:
Ahh! It sure is. I was making a big mistake...
Thank you very much. So it turns out that no fixes is needed for `RSpec/NamedSubject`.

Reply:
Though if you define `subject(:subject!)` it works